device_flow_test.py

```diff
@@ -0,0 +1,76 @@
+import os
+import pathlib
+
+import simple_ado
+import argparse
+import logging
+from auth_helper import AuthHelper
+from simple_ado import ADOException
+from simple_ado.exceptions import ADOHTTPException
+
+def main():
+    logger = logging.getLogger("test")
+    project_id = os.environ["SIMPLE_ADO_PROJECT_ID"]
+    repo_id = os.environ["SIMPLE_ADO_REPO_ID"]
+    token = os.environ["SIMPLE_ADO_BASE_TOKEN"]
+    username = os.environ["SIMPLE_ADO_USERNAME"]
+    tenant = os.environ["SIMPLE_ADO_TENANT"]
+    output = pathlib.Path(__file__).parent / pathlib.Path(outputDir) / pathlib.Path(repo_id + ".zip")
+    if not pathlib.Path.is_dir(pathlib.Path(output).parent):
+        pathlib.Path.mkdir(pathlib.Path(output).parent)
+    print("* Fetching the repo: " + repoUrlStr)
+    
+    try:
+        ah = AuthHelper()
+        token = ah.adoAuthenticate()
+
+        print("** Setting up ADOHTTPClient with " + tenant)
+        http = simple_ado.http_client.ADOHTTPClient(tenant=tenant,
+                    credentials=token,
+                    user_agent="test",
```

Comment:
Use a valid user agent. We want to be respectful when using the service.

Reply:
Do you have one in mind? Or just leave it unset so we get some kind of default?

Reply:
Something like `simple_ado/tests` would be fine.